### PR TITLE
[Improvement][BatchQuery] Batch query ProcessDefinitions belongs to need failover ProcessInstance.

### PR DESCRIPTION
**Pull request by BongBongBang**
## Purpose of the pull request

- Batch query ProcessDefinitions belongs to need failover ProcessInstance.
- This PR‘s implementation also fits in the [Refactor]Migrate some interface functions from ProcessServiceImpl to DAO, like [[Refactor] Migrate all taskInstance-related interface functions from ProcessServiceImpl #12505](https://github.com/apache/dolphinscheduler/issues/12469) 
- This PR is same as [[Improvement][BatchQuery] Batch query ProcessDefinitions belongs to need failover ProcessInstance. #12499](https://github.com/apache/dolphinscheduler/pull/12499) 

## Brief change log

[Improvement][BatchQuery] Batch query ProcessDefinitions belongs to need failover ProcessInstance.

## Verify this pull request

This pull request is already covered by existing tests, such as *(please describe tests)*.

### CODE REVIEW

File: dolphinscheduler-dao/src/main/resources/org/apache/dolphinscheduler/dao/mapper/ProcessDefinitionLogMapper.xml
Comment (caishunfeng) [inline dolphinscheduler-dao/src/main/resources/org/apache/dolphinscheduler/dao/mapper/ProcessDefinitionLogMapper.xml]: It should avoid the use `or` in query, because it will miss the index and query slowly.

**Review by caishunfeng — CHANGES_REQUESTED** [review]

File: dolphinscheduler-dao/src/main/resources/org/apache/dolphinscheduler/dao/mapper/ProcessDefinitionLogMapper.xml
Comment (BongBongBang) [reply dolphinscheduler-dao/src/main/resources/org/apache/dolphinscheduler/dao/mapper/ProcessDefinitionLogMapper.xml]: Ok, I took this change advice from wenjun. Let me check the sql explain. If it doesn't use index in reality, we have to query `ProcessDefinitionLog` one by one here.

**Review by caishunfeng — DISMISSED** [review]
LGTM if CI pass.

**Comment by BongBongBang** [comment]
Oops. I run ./mvnw spotless:check locally ,and there doesn't have violations...And after run ./.mvnw spotless:apply ,there has nothing changed.....How could i fix this...Any idea?

**Comment by BongBongBang** [comment]
。。。

**Comment by caishunfeng** [comment]
> Oops. I run ./mvnw spotless:check locally ,and there doesn't have violations...And after run ./.mvnw spotless:apply ,there has nothing changed.....How could i fix this...Any idea?

I approve to run the CI (Due to you are the first contributor)  and will check it after the CI finish.

**Comment by BongBongBang** [comment]
SonarCloud reported that `MasterFailoverService` constructor has 8 parameters which is more than 7.....I was blocked by the `TaskInstanceDao` that it's added after mine

**Comment by BongBongBang** [comment]
I found there are still three functions of `ProcessService` in `MasterFailoverService`. Let me migrate them to according `DAO` to remove the `ProcessService` parameter. Also, this change fits the Migrate issue [#12469 [https://github.com/apache/dolphinscheduler/issues/12469 ]](url)

**Comment by BongBongBang** [comment]
Let me remove `ProcessDefinitionDao` parameter from constructor and inject it

**Comment by caishunfeng** [comment]
I had approved to run CI (due to first time contributor )

**Comment by caishunfeng** [comment]
@BongBongBang please use spotless to format code style.
see code style  https://dolphinscheduler.apache.org/en-us/docs/dev/user_doc/contribute/development-environment-setup.html

**Comment by BongBongBang** [comment]
> @BongBongBang please use spotless to format code style. see code style https://dolphinscheduler.apache.org/en-us/docs/dev/user_doc/contribute/development-environment-setup.html

Actually, I do did that maven command before submission because of the failure about `spotless` before...... Maybe i forgot again....My fault.....

**Comment by BongBongBang** [comment]
Like chinese idiom, 好事多磨

**Comment by caishunfeng** [comment]
> > @BongBongBang please use spotless to format code style. see code style https://dolphinscheduler.apache.org/en-us/docs/dev/user_doc/contribute/development-environment-setup.html
> 
> Actually, I do did that maven command before submission because of the failure about `spotless` before...... Maybe i forgot again....My fault.....

You can set the pre-commit (see the doc above), and then it will auto spotless every commit.

**Review by caishunfeng — APPROVED** [review]
LGTM if CI pass

**Comment by caishunfeng** [comment]
Hi @BongBongBang , thanks for your contribution and welcome to join community 🎉 . Looking forward to your next contribution.

**Pull request merged**
